llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -6588,7 +6589,8 @@ SDValue TargetLowering::BuildUDIV(SDNode *N, SelectionDAG &DAG,
     if (VT.isVector())
       WideVT = EVT::getVectorVT(*DAG.getContext(), WideVT,
                                 VT.getVectorElementCount());
-    if (isOperationLegalOrCustom(ISD::MUL, WideVT)) {
+    if (isOperationLegalOrCustom(ISD::MUL, WideVT) ||
```

Comment:
I tried expanding to MULH/MUL_LOHI but the codegen is a bit worse, there are a few more instructions. Mul seem best.

Reply:
Would using `V_MUL_HI_U32` / `V_MUL_HI_S32` work?

https://www.amd.com/content/dam/amd/en/documents/instinct-tech-docs/instruction-set-architectures/amd-instinct-mi300-cdna3-instruction-set-architecture.pdf